src/GovUk.Education.ExploreEducationStatistics.Admin/Security/AuthorizationHandlers/ViewSpecificPublicationAuthorizationHandlers.cs

```diff
@@ -62,12 +63,29 @@ public HasRoleOnAnyChildReleaseAuthorizationHandler(ContentDbContext context)
         }
 
         public class HasOwnerRoleOnPublicationAuthorizationHandler
-            : HasRoleOnPublicationAuthorizationHandler<ViewSpecificPublicationRequirement>
+            : AuthorizationHandler<ViewSpecificPublicationRequirement, Publication>
         {
+            private readonly AuthorizationHandlerResourceRoleService _authorizationHandlerResourceRoleService;
+
             public HasOwnerRoleOnPublicationAuthorizationHandler(
-                IUserPublicationRoleRepository publicationRoleRepository) : base(publicationRoleRepository, context =>
-                ContainPublicationOwnerRole(context.Roles))
+                AuthorizationHandlerResourceRoleService authorizationHandlerResourceRoleService)
             {
+                _authorizationHandlerResourceRoleService = authorizationHandlerResourceRoleService;
+            }
+
+            protected override async Task HandleRequirementAsync(
+                AuthorizationHandlerContext context, 
+                ViewSpecificPublicationRequirement requirement,
+                Publication publication)
+            {
+                if (await _authorizationHandlerResourceRoleService
+                        .HasRolesOnPublication(
+                            context.User.GetUserId(),
+                            publication.Id,
+                            Owner))
```

Comment:
`ViewSpecificReleaseAuthorizationHandlers` allows publication owners and publication approvers - I think I'd expect this to do the same?

Reply:
Totally agree, and surprised this didn't cause issues when I was doing my testing!  Updated.